$OpenBSD: patch-src_libstd_sys_unix_thread_rs,v 1.1 2018/03/13 08:08:03 landry Exp $
Don't change permission on stack for creating a stack_guard page: we already have such thing.
Index: src/libstd/sys/unix/thread.rs
--- src/libstd/sys/unix/thread.rs.orig
+++ src/libstd/sys/unix/thread.rs
@@ -269,7 +269,7 @@ pub mod guard {
                 as *mut libc::c_void;
         }
 
-        if cfg!(target_os = "linux") {
+        if cfg!(target_os = "linux") || cfg!(target_os = "openbsd") {
             // Linux doesn't allocate the whole stack right away, and
             // the kernel has its own stack-guard mechanism to fault
             // when growing too close to an existing mapping.  If we map
